src/barfTurn.ts

```diff
@@ -0,0 +1,369 @@
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("lastVoteMonsterTurn") === totalTurnsPlayed();
+    },
+    spendsTurn: false,
+  },
+  {
+    name: "Digitize Wanderer",
+    available: () => Counter.get("Digitize Monster") <= 0,
+    execute: () => {
+      const isEmbezzler = SourceTerminal.getDigitizeMonster() === embezzler;
+      const start = get("_sourceTerminalDigitizeMonsterCount");
+
+      const shouldGoUnderwater =
+        isEmbezzler &&
+        !get("_envyfishEggUsed") &&
+        myLevel() >= 11 &&
+        (getModifier("Adventure Underwater") ||
+          waterBreathingEquipment.some((item) => have(item))) &&
+        (getModifier("Underwater Familiar") ||
+          familiarWaterBreathingEquipment.some((item) => have(item))) &&
+        (have($effect`Fishy`) || (have($item`fishy pipe`) && !get("_fishyPipeUsed"))) &&
+        !have($item`envyfish egg`) &&
+        canAdventure($location`The Briny Deeps`);
+      const targetLocation = shouldGoUnderwater
+        ? $location`The Briny Deeps`
+        : determineDraggableZoneAndEnsureAccess();
+
+      isEmbezzler ? embezzlerPrep() : freeFightPrep();
+      adventureMacroAuto(targetLocation, Macro.basicCombat());
+      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+    },
+    spendsTurn: () => !SourceTerminal.getDigitizeMonster()?.attributes.includes("FREE"),
+  },
+  {
+    name: "Guaranteed Kramco",
+    available: () => sober() && kramcoGuaranteed(),
+    execute: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return !kramcoGuaranteed();
+    },
+    spendsTurn: false,
+  },
+  {
+    name: "Void Monster",
+    available: () =>
+      sober() &&
+      have($item`cursed magnifying glass`) &&
+      get("cursedMagnifyingGlassCount") === 13 &&
+      get("_voidFreeFights") < 5,
+    execute: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`cursed magnifying glass` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("cursedMagnifyingGlassCount") === 0;
+    },
+    spendsTurn: false,
+  },
+  {
+    name: "Envyfish Egg",
+    available: () =>
+      have($item`envyfish egg`) && get("envyfishMonster") === embezzler && !get("_envyfishEggUsed"),
+    execute: () => {
+      embezzlerPrep();
+      withMacro(Macro.meatKill(), () => use($item`envyfish egg`), true);
+      return get("_envyfishEggUsed");
+    },
+    spendsTurn: true,
+  },
+  {
+    name: "Spit Acid",
+    available: () =>
+      sober() && have($item`Jurassic Parka`) && !have($effect`Everything Looks Yellow`),
+    execute: () => {
+      const jellyfishing = have($familiar`Space Jellyfish`) && realmAvailable("stench");
+      const familiarChoice = jellyfishing ? $familiar`Space Jellyfish` : freeFightFamiliar(true);
```

Comment:
Just a reminder (it's probably already in your list of considerations) that after the small revamp, `jellyfishing` should be conditioned upon `familiarChoice`, not this other way around (i.e. `jellyfishing = familiarChoice === $familiar'Space Jellyfish'`)

Reply:
I wrote draft logic that essentially went
can we jellyfish? if so, tell the function that our location is pirate barges
did we get jellyfish? if so, our location is pirate barges! else, our location is a draggable zone.

This has a bit of a flaw: what if the value of placing a wanderer would make a different familair beat the jellyfish? but also that is so marginal I cannot be arsed to care. And I'm not even british!